sklearn/tests/test_docstring_parameters.py

```diff
@@ -0,0 +1,166 @@
+# Authors: Alexandre Gramfort <alexandre.gramfort@inria.fr>
+#          Raghav RV <rvraghav93@gmail.com>
+# License: BSD 3 clause
+
+from __future__ import print_function
+
+from nose.tools import assert_true
+import inspect
+import warnings
+import importlib
+
+from pkgutil import walk_packages
+from inspect import getsource
+
+import sklearn
+from sklearn.base import signature
+from sklearn.utils.testing import SkipTest
+from sklearn.utils.testing import check_parameters_match, get_func_name
+
+_DOC_SPECIAL_MEMBERS = ('__contains__', '__getitem__', '__iter__', '__len__',
+                        '__call__', '__add__', '__sub__', '__mul__', '__div__',
+                        '__neg__', '__hash__')
+
+PUBLIC_MODULES = [
```

Comment:
@agramfort I thought fixing this up in a single issue is going to blow this issue up. My experience gives me a bad gut feeling when a PR goes over 1.5k lines :p 

If it's fine by you, like @ogrisel suggests, I'll finish up pending comments and add an issue + comment here. I'll fix it up in a subsequent PR?